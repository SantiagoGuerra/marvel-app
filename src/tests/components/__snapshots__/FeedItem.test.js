const React = require('react');
const { shallow } = require('enzyme');
const { default: FeedItem } = require('../../../components/FeedItem');

describe('Test <FeedItem /> Component', () => {
  const name = 'Hulk';
  const thumbnail = {
    path: 'https://...',
    extension: 'png',
  };

  const wrapper = shallow(<FeedItem name={name} thumbnail={thumbnail} />);

  test('should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have data as parameter', () => {
    const wrapper = shallow(<FeedItem name={name} thumbnail={thumbnail} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a component called Text', () => {
    const Text = wrapper.find('Text');

    expect(Text.exists()).toBe(true);
  });
});
