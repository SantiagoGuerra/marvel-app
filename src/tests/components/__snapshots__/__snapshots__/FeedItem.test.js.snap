// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FeedItem /> Component should have data as parameter 1`] = `
<Styled(Box)
  animation={
    Array [
      "fadeIn",
      "slideDown",
    ]
  }
  class="feed-item"
>
  <img
    alt="Hulk"
    className="img"
    src="https://....png"
  />
  <Text
    className="text"
    level={1}
  >
    Hulk
  </Text>
</Styled(Box)>
`;

exports[`Test <FeedItem /> Component should render the component correctly 1`] = `
<Styled(Box)
  animation={
    Array [
      "fadeIn",
      "slideDown",
    ]
  }
  class="feed-item"
>
  <img
    alt="Hulk"
    className="img"
    src="https://....png"
  />
  <Text
    className="text"
    level={1}
  >
    Hulk
  </Text>
</Styled(Box)>
`;
